modules/tinymce/src/core/main/ts/selection/SetSelectionContent.ts

```diff
@@ -102,7 +102,7 @@ const cleanContent = (editor: Editor, args: SetSelectionContentArgs) => {
     const contextBlock = editor.dom.getParent(rng.commonAncestorContainer, editor.dom.isBlock);
     const contextArgs = contextBlock ? { context: contextBlock.nodeName.toLowerCase() } : { };
 
-    const node = editor.parser.parse(args.content, { isRootContent: true, forced_root_block: false, ...contextArgs, ...args });
```

Comment:
I believe this `isRootContent: true` was incorrect, as we're parsing content that will be inserted not into the root, but at a specific point of selection. See the `isAtEdgeOfBlock` function in `DomParser.ts` for where we use this setting.

Reply:
This shouldn't have made any difference as the `forced_root_block: false` means `isRootContent` should be ignored. That likely means there's a bug somewhere else in your new implementation. I do agree `isRootContent: true` shouldn't have been here though.